### ISSUE TITLE
Feature request: Support running without root

### ISSUE THREAD

**Issue by catern** [saschagrunert/kubernix#16]
If Nix is already installed, there shouldn't be a need to run as root.

**Comment by saschagrunert** [comment]
Hey @catern, thank you for the request. Do you want to contribute to this via a PR? :)

**Comment by catern** [comment]
I don't myself have a need to run Kubernetes - I don't actually know if Kubernetes can even run without root. But I help run a collection of Unix machines where the users have access to Nix but no root; if Kubernix was usable by non-root, they could use it. Just letting you know about this use case (where Nix is available but root is not).

**Comment by saschagrunert** [comment]
Yeah, there are efforts in rootless Kubernetes, but since CRI-O does not support it right now out of the box we will stick to the root-mode.

**Comment by cwegener** [comment]
After 5 years in alpha, kubelet-in-UserNS will promote to beta in 1.37
This could be a good opportunity to support this in kubernix

**Comment by saschagrunert** [comment]
v0.3.5 has rootless support!
